plugin.go

```diff
@@ -57,77 +56,29 @@ func (p *Plugin) DiscoverActions(_ context.Context) ([]*action.Action, error) {
 			term = rt.Term()
 		}
 
-		u, err := createUpdateAction(p.k, ci)
+		cmd, err := getUpdateCmd()
 		if err != nil {
 			return err
 		}
+
+		u := &updateAction{
+			k:             p.k,
+			credentials:   ci,
+			sudoCmd:       cmd,
+			cfg:           getUpdateConfig(),
+			externalCfg:   input.Opt("config").(string),
+			targetVersion: input.Opt("target").(string),
+		}
 		u.SetLogger(log)
 		u.SetTerm(term)
 
-		return runUpdate(u)
-	}))
-	return []*action.Action{a}, nil
-}
-
-// runUpdate command entrypoint.
-func runUpdate(u *updateAction) error {
-	// Wrapper to conclude errors.
-	if err := runCommands(u); err != nil {
-		u.exitWithError()
-		return err
-	}
-
-	return nil
-}
-
-// runCommands run commands one by one.
-func runCommands(u *updateAction) error {
-	version := launchr.Version()
-	u.Term().Info().Printfln("Starting %s installation...", version.Name)
-
-	currOS, arch, err := u.initVars()
-	if err != nil {
-		return err
-	}
-
-	// Check the validity of the credentials.
-	if err = u.validateCredentials(); err != nil {
-		return err
-	}
-
-	// Get value of Stable Release.
-	stableRelease, err := u.getStableRelease()
-	if err != nil {
-		return err
-	}
-
-	if isUpToDate(stableRelease) {
-		u.Term().Printfln("Current version of %s is up to date.", version.Name)
-		return nil
-	}
-
-	// Format the URL with the determined 'os', 'arch' and 'extension' values.
-	u.c.URL = fmt.Sprintf(binPathMask, baseURL, stableRelease, currOS, arch, u.ext)
-	u.Term().Printfln("Downloading file: %s", u.c.URL)
-
-	// Download file to the temp folder.
-	if err = u.downloadFile(); err != nil {
-		return err
-	}
-
-	u.Log().Debug("binary path", "path", u.fPath)
+		err = u.doRun()
+		if err != nil {
+			u.Term().Error().Println("Update failed")
+			u.cleanup()
+		}
```

Comment:
Looks like it does not work as is for public repositories:
```
➜ cat test.yaml                                 
repository_url: "https://raw.githubusercontent.com/davidferlay/launchrctl-public-repo-test/main/"
latest_stable: "stable_release.txt"
bin_mask: "%s/%s/plasmactl_%s_%s%s"

➜ plx update --config test.yaml
 INFO  Starting plasmactl installation...
Enter passphrase to unlock the keyring: 
 INFO  Enter credentials for https://raw.githubusercontent.com/davidferlay/launchrctl-public-repo-test/main/
Username: 
Password: 
```
Credentials are not required in this case, so it should not prompt for it